# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/agnostic/bytestreams/AnnexB.h
# Commit: 6f7b5bf9db07
# Full Hash: 6f7b5bf9db07f4b0e5ff088257b10772f1486340
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-06 15:27:33
# Regressor Bug: 1852333
# File Overlap Count: 4
# Description:
#   Bug 1852333 - part2 : detect config change for HEVC stream. r=media-playback-reviewers,padenot
#   
#   This patch detects inband change for the HEVC playback so that we can
#   recreate decoder if the new config is different from the previous one.
#   
# ==============================================================================

diff -r 1de872081cfc -r 6f7b5bf9db07 dom/media/platforms/agnostic/bytestreams/AnnexB.h
--- a/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Thu Oct 05 21:07:14 2023 +0000
+++ b/dom/media/platforms/agnostic/bytestreams/AnnexB.h	Thu Oct 05 21:07:15 2023 +0000
@@ -43,6 +43,8 @@
   static bool ConvertSampleToAVCC(
       mozilla::MediaRawData* aSample,
       const RefPtr<mozilla::MediaByteBuffer>& aAVCCHeader = nullptr);
+  // Convert a sample from Annex B to HVCC. An HVCC extradata must not be set.
+  static bool ConvertSampleToHVCC(mozilla::MediaRawData* aSample);
 
   // Covert sample to 4 bytes NALU byte stream.
   static mozilla::Result<mozilla::Ok, nsresult> ConvertAVCCTo4BytesAVCC(